core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -659,13 +685,21 @@ class GroupMetadataManager(val brokerId: Int,
       config.offsetsTopicNumPartitions
   }
 
-  private def getMessageFormatVersionAndTimestamp(partition: Int): (Byte, Long) = {
+  /**
+   * Check if the replica is local and return the message format version and timestamp
+   *
+   * @param partition Partition of GroupMetadataTopic
+   * @return Option[(MessageFormatVersion, TimeStamp)]
+   */
+  private def getMessageFormatVersionAndTimestamp(partition: Int): Option[(Byte, Long)] = {
     val groupMetadataTopicAndPartition = new TopicAndPartition(Topic.GroupMetadataTopicName, partition)
-    val messageFormatVersion = replicaManager.getMessageFormatVersion(groupMetadataTopicAndPartition).getOrElse {
-      throw new IllegalArgumentException(s"Message format version for partition $groupMetadataTopicPartitionCount not found")
+    val messageFormatVersionOpt = replicaManager.getMessageFormatVersion(groupMetadataTopicAndPartition)
+    messageFormatVersionOpt match {
+      case None => None
+      case Some(messageFormatVersion) =>
+        val timestamp = if (messageFormatVersion == Message.MagicValue_V0) Message.NoTimestamp else time.milliseconds()
+        Some(messageFormatVersion, timestamp)
```

Comment:
You can use `map` instead of the pattern match as you are changing the value in case it's `Some`.